src/state/lists/hooks.ts

```diff
@@ -46,8 +48,57 @@ export function listToTokenMap(list: TokenList | null, useCache = true): TokenAd
   return map
 }
 
-export function useAllLists(): AppState['lists']['byUrl'] {
-  return useSelector<AppState, AppState['lists']['byUrl']>(state => state.lists.byUrl)
+const selectListByUrl = (state: AppState): AppState['lists']['byUrl'] => state.lists.byUrl
+
+export function useAllLists() {
+  return useSelector(selectListByUrl)
+}
+
+const selectTokensBySymbol = createSelector(selectListByUrl, lists => {
+  const allTokens = new Map<string, TokenInfo>()
+  for (const key in lists) {
+    lists[key]?.current?.tokens?.forEach(token => {
+      allTokens.set(token.symbol.toUpperCase(), token)
+    })
+  }
+  return allTokens
+})
+
+const selectTokensByAddress = createSelector(selectListByUrl, lists => {
+  const allTokensByChain = new Map<ChainId, Map<string, TokenInfo>>()
+  for (const key in lists) {
+    lists[key]?.current?.tokens?.forEach(token => {
+      if (allTokensByChain.has(token.chainId)) {
+        const tokenMap = allTokensByChain.get(token.chainId)
+        if (tokenMap) {
+          tokenMap.set(token.address, token)
+        }
+      } else {
+        allTokensByChain.set(token.chainId, new Map<string, TokenInfo>())
+      }
+    })
+  }
+  for (const key of allTokensByChain.keys()) {
+    const network = NETWORK_DETAIL[key]
+    if (network) {
+      const tokenMap = allTokensByChain.get(key)
+      tokenMap?.set('0x0000000000000000000000000000000000000000', {
+        chainId: key,
+        address: '0x0000000000000000000000000000000000000000',
```

Comment:
We could use `AddressZero` constant here too.

Reply:
Done 👍🏼